packages/web/src/components/ui/heading.tsx

```diff
@@ -1,36 +1,72 @@
 import * as React from 'react';
 
 import { cn } from '@/lib/utils';
+import { Info } from 'lucide-react';
+import { z } from 'zod';
 
 const className =
-  'flex w-full rounded-md border border-transparent bg-transparent px-1 py-1 transition-colors hover:border-input hover:shadow-sm focus-visible:shadow-md focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50';
+  'flex w-full break-all whitespace-normal rounded-md border border-transparent bg-transparent px-1 py-1 transition-colors hover:border-input hover:shadow-sm focus-visible:shadow-md focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50';
 
 export function EditableH1(props: {
   text: string;
   className?: string;
   onUpdated: (text: string) => void;
 }) {
-  const ref = React.useRef<HTMLHeadingElement | null>(null);
+  const ref = React.useRef<HTMLHeadingElement>(null);
+  const [isMaxHeadingLengthExceeded, setIsMaxHeadingLengthExceeded] =
+    React.useState<boolean>(false);
+
+  const maxHeadingLength = 44;
+  const headingSchema = z.string().max(maxHeadingLength, {
```

Comment:
A schema for the title is already defined in the shared package, so we should be able to import and reuse that here rather than defining a new one.

Reply:
Done!